File: leet code practice/Final Prices With a Special Discount in a Shop.js
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
    var res=[]
    for(let i=0;i<prices.length;i++){
        var finalprice=prices[i]
        for(let j=i+1;j<prices.length;j++){
            if(prices[j]<=prices[i]){
                finalprice=prices[i]-prices[j]
                break;
            }
        }res.push(finalprice)
    }
    return res
};

File: leet code practice/Final Prices With a Special Discount in a Shop.py
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        res=[]
        for i in range(len(prices)):
            finalprice=prices[i]
            for j in range(i+1,len(prices)):
                if prices[j]<=prices[i]:
                    finalprice=prices[i]-prices[j]
                    break
            res.append(finalprice)
        return res